### PR TITLE
[Component with its own class name as service type] LPS-200419 Refactor Adapter component 

### PR DESCRIPTION
**Pull request by Valmir-unicap**

### CODE REVIEW

**Comment by Valmir-unicap** [comment]
ci:test:sf

**Review by MrJohn1911 — APPROVED** [review]

**Comment by Valmir-unicap** [comment]
I pushed just to adjust the name of the submission suggested by @MrJohn1911

File: modules/dxp/apps/portal-search-tuning/portal-search-tuning-synonyms-web/src/main/java/com/liferay/portal/search/tuning/synonyms/web/internal/storage/SynonymSetStorageAdapter.java [hunk 1]
```diff
@@ -15,28 +15,36 @@
...
 /**
  * @author Bryan Engler
  */
-@Component(service = SynonymSetStorageAdapter.class)
 public class SynonymSetStorageAdapter {
 
+	public static SynonymSetStorageAdapter getInstance(
+		SynonymSetIndexWriter synonymSetIndexWriter,
+		SynonymSetJSONStorageHelper synonymSetJSONStorageHelper) {
+
```
Comment (rafaprax) [inline modules/dxp/apps/portal-search-tuning/portal-search-tuning-synonyms-web/src/main/java/com/liferay/portal/search/tuning/synonyms/web/internal/storage/SynonymSetStorageAdapter.java:26]: make the singleton thread safe or transform it into a Util class

**Review by rafaprax — CHANGES_REQUESTED** [review]

**Comment by rafaprax** [comment]
and fix conflicts

**Pull request closed without merging**